### ISSUE TITLE
Showcase pinned repositories

### ISSUE THREAD

**Issue by PavlikPolivka** [github/personal-website#14]
Hi. 
I am new to Jekyll and this project maked it easy for me to understand how stuff works.

I have a question about the repositories showcase
```
{% for repository in site.github.public_repositories limit: 9 %}
```
Is it possible to iterate over pinned repositories? As someone with a lot of repositories I would like to showcase 9 one the important ones, not just first 9 as those are mostly forks of other open source projects. 

Thanks.

**Comment by ruslanys** [comment]
Agree. I do want to be able to exclude forks first of all. Also, I would like to show repositories sorted by stars desc. Or, at least, show only pinned ones.

**Comment by MarkWaldron** [comment]
It appears to be an issue with the plugin: jekyll-github-metadata. It only gives you access to a limited number of options. https://help.github.com/en/articles/repository-metadata-on-github-pages#using-repository-metadata-locally

**Comment by PavlikPolivka** [comment]
I was able to exclude forks and sort by stars.

```
  {% assign public_repositories = site.github.public_repositories | where:'fork', false | sort: 'stargazers_count' | reverse %}
  {% for repository in public_repositories limit: 9 %}
```

Not sure if I should submit this as PR.

**Comment by ruslanys** [comment]
@PavlikPolivka Perfect! Thank you!

**Comment by Ullaakut** [comment]
It's also possible to manually filter them out by name, which is a bit more tedious than @PavlikPolivka's solution but lets you show repositories that don't necessarily have many stars:

```html
  {% for repository in site.github.public_repositories limit: 99 %}
    {% if repository.name == "myrepo" %}
    <div class="col-sm-6 col-md-12 col-lg-6 col-xl-4 mb-3">
      {% include repo-card.html %}
    </div>
    {% endif %}
  {% endfor %}
```

This would only show repositories named `myrepo`. Chaining multiple if statements like this lets you choose the exact repositories you want to showcase.

**Comment by diev** [comment]
@Ullaakut 
> Chaining multiple if statements like this lets you choose the exact repositories you want to showcase.

Use *case* instead of multiple *if*:

```
{% case repository.name %}
  {% when "myrepo" %}
    ...
  {% when "myrepo2" %}
    ...
{% endcase %}

```

But for this case there is something better:

```
{% if "myrepo, finerepo, goodrepo, ..." contains repository.name %}
  ...
{% endif %}

```

Of course, the last case will be wrong if `repositary.name == "repo"`.

**Comment by kunicmarko20** [comment]
Ended up adding:

```
repositories:
  - SonataAnnotationBundle
```

to `_config.yml` file and then in `projcets.html`:

```
  {% for repository in site.github.public_repositories limit: 9 %}
    {% if site.repositories contains repository.name %}
      <div class="col-sm-6 col-md-12 col-lg-6 col-xl-4 mb-3">
        {% include repo-card.html %}
      </div>
    {% endif %}
  {% endfor %}
```

**Comment by PavlikPolivka** [comment]
Should the limit be still there? This way you can only filter from first 9
repositories.

**Comment by kunicmarko20** [comment]
@PavlikPolivka it was edited but I guess you got the old version in the email, the limit should be removed.

**Comment by brandonrosage** [comment]
My understanding is that forked repositories are often projects that are desirable to share. So I'm initially reluctant to add complexity (and confusion, potentially) by filtering out a subset of a user's public repositories.

A lighter touch, I think, could be adding this as a boolean option in the `_config.yml` file:

```
hide_forks: true
```

What do y'all think?

**Comment by mudlord** [comment]
I still think the idea of filtering projects to ones that you consider most important is a good one. I guess people could make a separate page listing all their github repositories and on the index page only list ones you consider important.

**Comment by aadibajpai** [comment]
Is there a way to also list external repositories that you have contributed to but do not own? We can list them in pinned repositories but I'd like to list them in my website as well.

**Comment by brandonrosage** [comment]
I should mention here that we merged a PR today, #61, that supports excluding forks, alongside [a handful of other quick customizations](https://github.com/github/personal-website#projects) to the "My Projects" listing.

**Comment by brandonrosage** [comment]
I'm investigating ways to access a "pinned" attribute for a user's repositories. 👍

**Comment by aadibajpai** [comment]
@brandonrosage did you find a way to access "pinned" repositories or even repositories from organizations that a user may be part of?

**Comment by OlivierJM** [comment]
> @brandonrosage did you find a way to access "pinned" repositories or even repositories from organizations that a user may be part of?

I found this https://github.com/SUPERCILEX/personal-website/commit/5367c2b02ffe380190da0f91e9adb7b1fcef12e9 from @SUPERCILEX It might help you add dynamic repos

**Comment by SUPERCILEX** [comment]
I've made some improvements since then so I would recommend looking at https://github.com/SUPERCILEX/personal-website/compare/b5de3764ae9d54e5aaae40b7ed62ff5d4f186657...ca987a5fd35a42e1769de8145e1a7019fc402762.
